R/check_result.R

```diff
@@ -5,19 +5,17 @@
 #' If the user result exactly matches a known \code{result}, \code{check_result}
 #' returns the matching message value.
 #'
-#' @param results A \code{\link{results}} object that contains possible \code{\link{result}} values to compare against.
 #' @param correct A character string to display if the student answer matches
 #'   a known answer.
 #'   This character string will be run through \code{glue::\link[glue]{glue_data}} with \code{list(correct = TRUE, message = "<result message>")}. where message is the matched result message.
 #' @param incorrect A character string to display if the student answer matches
 #'   a known answer.
 #'   This character string will be run through \code{glue::\link[glue]{glue_data}} with \code{list(correct = FALSE, message = "<result message>")}. where message is the matched result message.
-#' @param empty_msg A character string to display as a message if the user code is NULL.
+#' @template grader_args
+#' @template learnr_args
 #' @param ... ignored
-#' @param user (Optional) student code to check against the \code{results} surrounded
-#'   by \code{quote()}, \code{rlang::quo()}, or provided as a character string.
 #'
-#' @return a \code{grader_result} structure from \code{\link{result}} containing a formatted \code{correct} or \code{incorrect} message.
+#' @return a \code{grader_graded} structure from \code{\link{result}} containing a formatted \code{correct} or \code{incorrect} message and whether or not a match was found, \code{matched}.
```

Comment:
`grader_graded` does not return `matched`.  `matched` is a special parameter provided to the `correct` and `incorrect` strings when glued.